packages/wallet/src/services/util/coldObservableProvider.ts

```diff
@@ -49,10 +56,24 @@ export const coldObservableProvider = <T>({
                     )
               )
             )
-          ).pipe(retryBackoff(retryBackoffConfig))
+          ).pipe(
+            retryBackoff({
+              ...retryBackoffConfig,
+              shouldRetry: (error) => {
+                if (retryBackoffConfig.shouldRetry && !retryBackoffConfig.shouldRetry(error)) return false;
+
+                if (error instanceof InvalidStringError) {
+                  onFatalError?.(error);
+                  cancelOnFatalError$.next(true);
+                }
+
+                return true;
+              }
+            })
+          )
         ),
         distinctUntilChanged(equals),
-        takeUntil(cancel$)
+        takeUntil(internalCancel$)
       )
       .subscribe(subscriber);
     return () => sub.unsubscribe();
```

Comment:
I think it would be good to also complete the `cancelOnFatalError$` subject in the unsubscribe callback:
```diff
-     return () => sub.unsubscribe();
+     return () => { sub.unsubscribe(); cancelOnFatalError$.complete(); };

Reply:
Nice catch!

Reply:
dcda82612 should solve this